### DISCUSSION TITLE
Detekt 2.0 configurations

### DISCUSSION THREAD

**Discussion by picklebento** [detekt/detekt#3784] [Ideas]
This is a thread to discuss those configuration names that need to be fixed in a backward incompatible way.

ForbiddenClassName > forbiddenName: This should be renamed to `forbiddenNames`

Indentation > continuationIndentSize: unused

FunctionParameterNaming > ignoreOverriddenFunctions: deprecated
MemberNameEqualsClassName > ignoreOverriddenFunction: deprecated
LongParameterList > ignoreOverriddenFunction: deprecated
EmptyFunctionBlock > ignoreOverriddenFunctions: deprecated

**Comment by marschwar** [comment]
ForbiddenImport > imports should be `forbiddenImports`
FunctionOnlyReturningConstant > excludeAnnotatedFunctions should be `excludedAnnotatedWith` (or something similar)
FunctionOnlyReturningConstant > excludedFunctions to something more meaningful. Maybe even convert to a list.
MagicNumber > ignoreNumbers should be `ignoredNumbers`
WildcardImport > excludeImports should be `excludedImports`

**Comment by marschwar** [comment]
`ThrowingExceptionsWithoutMessageOrCause`, `TooGenericExceptionCaught` and `TooGenericExceptionThrown` all define exceptions that should be reported. The config property should be the same in my opinion (currently: `exceptionNames` vs. `exceptions`). `SwallowedException` allows to configure exceptions that should be ignored. The property now is `ignoredExceptionTypes`. It should be consistent with those above.
